portindex2json/portindex2json.tcl

```diff
@@ -12,6 +12,78 @@
 
 package require json::write
 
+proc format_strings {strings} {
+    if {[llength $strings] > 1} {
+        foreach element $strings {
+            lappend new_strings "\"$element\""
+        }
+    } else {
+        set new_strings [list "\"$strings\""]
+    }
+    return $new_strings
+}
+
+proc get_maintainer_array {maintainer} {
+    set detail_list [split $maintainer @]
+    if {[string first : [lindex $detail_list 0]] != -1} {
+        set email_list [split [lindex $detail_list 0] :]
+        set name [string trimright [lindex $email_list 1]]
+        set email(name) "\"$name\""
+        set email(domain) "\"[lindex $email_list 0]\""
+        ::json::write indented false
+        set email_subobject [::json::write object {*}[array get email]]
+        ::json::write indented true
+    } else {
+        set name [string trimright [lindex $detail_list 0]]
+        set email(name) "\"$name\""
+        set email(domain) "\"macports.org\""
+        ::json::write indented false
+        set email_subobject [::json::write object {*}[array get email]]
+        ::json::write indented true
+    }
+
+    set maintainer_array(github) "\"[lindex $detail_list 1]\""
+    set maintainer_array(email) $email_subobject
+
+    return [array get maintainer_array]
+}
+
+proc parse_maintainers {maintainers} {
+    set ismaintainer 0
+    foreach maintainer $maintainers {
+        if {$maintainer == "openmaintainer"} {
+            continue
+        } elseif {$maintainer == "nomaintainer"} {
+            continue
+        } else {
+            lappend maintainer_subobjects [::json::write object {*}[get_maintainer_array $maintainer]]
+            set ismaintainer 1
+        }
+    }
+    if {$ismaintainer == 1} {
+        set maintainers_list [::json::write array {*}$maintainer_subobjects]
+        unset maintainer_subobjects
+        return $maintainers_list
+    } else {
+        return "\[\]"
+    }
+}
+
+proc is_closedmaintainer {maintainers} {
+    set closedmaintainer true
+    foreach maintainer $maintainers {
+        if {$maintainer == "openmaintainer" || $maintainer == "nomaintainer"} {
+            set closedmaintainer false
+        }
+    }
+    return $closedmaintainer
+}
+
+proc remove_extra_braces {text} {
+    return [regsub -all {[\{\}]} $text ""]
```

Comment:
We should probably just stop fixing that after the fact. The right thing to do in the Portfile is write `long_description {*}${description}`.

That said, if $long_description is length 1 then it's safe to use `[lindex $long_description 0]`.

Reply:
Now that I saw the source of the problem, I agree that fixing it here is kind of the wrong place to perform any fixes.

However, we would need to fix nearly 4000 portfiles if we follow your suggestion. Definitely off-topic for this PR, but my naive question is: Tcl is eventually able to print the string to the screen without all the wrong quoting. Is there really no way to write the properly expanded string to PortIndex automatically if we add some fixes to the base?

Reply:
> Tcl is eventually able to print the string to the screen without all the wrong quoting.

It's not "wrong" quoting, it's precisely the quoting needed to define the list as it was given. If you want to join all the list elements together into one string, you use `join`. Tcl doesn't just decide to do this on its own; have a look at what action_info does.

> Is there really no way to write the properly expanded string to PortIndex automatically if we add some fixes to the base?

It would be very strange to have the value of $long_description in the Portfile be different to the value you get from reading the PortIndex. Think about what this would mean: you would get one answer from `mportinfo` and another from `mportlookup` (both return the same PortInfo array, but the former does it by parsing the Portfile and the latter gets it from the PortIndex.)